code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -163,6 +164,84 @@ func (ssc *StorageSmartContract) GetBlobbersHandler(
 	return sns, nil
 }
 
+// GetAllocationBlobbersHandler returns list of all blobbers alive (e.g. excluding
+// blobbers with zero capacity).
+func (ssc *StorageSmartContract) GetAllocationBlobbersHandler(
+	ctx context.Context,
+	params url.Values, balances cstate.StateContextI) (interface{}, error) {
+
+	var creationDate = common.Timestamp(time.Now().Unix())
+
+	allocData := params.Get("allocation_data")
+	var request newAllocationRequest
+	if err := request.decode([]byte(allocData)); err != nil {
+		return "", common.NewErrInternal("can't decode allocation request", err.Error())
+	}
+
+	var sa = request.storageAllocation()
+	if balances.GetEventDB() == nil {
+		return nil, errors.New("events db is not initialised")
+	}
+
+	blobbers, err := getBlobbersFromQuery(params, balances.GetEventDB())
+	if err != nil || len(blobbers) == 0 {
+		return nil, fmt.Errorf("no blobbers found %v", err)
+	}
+
+	var sns StorageNodes
+	for _, blobber := range blobbers {
+		sn, err := blobberTableToStorageNode(blobber)
+		if err != nil {
+			return ssc.GetBlobbersHandlerDeprecated(ctx, params, balances)
+		}
+		sns.Nodes.add(&sn)
+	}
+
+	var size = sa.DataShards + sa.ParityShards
+
+	var bSize = (sa.Size + int64(size-1)) / int64(size)
+
+	var list = sa.filterBlobbers(sns.Nodes.copy(), creationDate,
```

Comment:
For new allocation request we only want to return a list of blobber ids, not the actual objects.

Reply:
Making this change

Reply:
Why not modify filterBlobbers to return a list of ids.

In [filterBobbers](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/models.go#L740) you can change
```go
list[i] = b
```
to
```go
list[i] = b.ID
```
Maybe make a copy `filterBlobberFunc` with this change, and mark the old version as depreciated as It is only needed until https://github.com/0chain/0chain/pull/684 is merged.

Reply:
Why can't you do the work that filterBlobber is doing as part of the SQL query?

Have a `select blobber_id from blobber where` clause that basically does the same work as filterBlobbers. Then you just take the first `N` ids returned by the query and do not have to bother with `filterBlobbers` at all.

Reply:
something like
```sql
select blobber_id from blobbers where 
  MaxOfferDuration < due and 
  ReadPrice > minRP and ReadPrice < maxRP and 
  WritePrice > minWP and WritePrice < maxWP and 
  Capacity < size and ChallengeCompletionTime < MaxCCT and 
  LastHeathCheck <= now-MaxCheckTime and
  TotalStake - CommitedStake>  size*writePrice
```

Reply:
`MaxOfferDuration >= due` corrected